### ISSUE TITLE
data augmentation applied only to images and not GT?

### ISSUE THREAD

**Issue by glhr** [metahexane/ssma_pytorch#1]
I might be missing something obvious, but the random crop/rotation/resize applied to the input RGB and depth images in `loader.py`:
```
pilRGB = self.data_augmentation(pilRGB)
pilDep = self.data_augmentation(pilDep)
```
isn't applied to `modGT`. Doesn't that cause misalignment between the GT labels and modified input images?

**Comment by glhr** [comment]
Applying random data augmentation to each modality separately like this will also cause mis-alignment between the RGB and depth images...

**Comment by metahexane** [comment]
You are correct, the data augmentation function should apply the same augmentation to all modalities. Furthermore, the GT should indeed also be augmented, but only if the modalities undergo a spatial transformation, such as a crop, resize etc. Applying Color Jitter to the GT will not help the network. Also, when resizing the GT for augmentation, the nearest neighbour method should be used as to not change the labels of the pixels.

I am now quite busy with my thesis and have not enough time to pick this up. You can work the solution out if you want and open up a PR (I will add you to the contributors if u agree).
